### PR TITLE
CMStatusReportingComponent: A resource to configure the CM Status Reporting Component

### PR DESCRIPTION
**Pull request by NEllis280**
#### This Pull Request (PR) fixes the following issues
None

#### Task list
- [x] Added an entry to the change log under the Unreleased section of the
      file CHANGELOG.md. Entry should say what was changed and how that
      affects users (if applicable), and reference the issue being resolved
      (if applicable).
- [x] Resource documentation added/updated in README.md.
- [x] Resource parameter descriptions added/updated in README.md, schema.mof
      and comment-based help.
- [x] Comment-based help added/updated.
- [x] Localization strings added/updated in all localization files as appropriate.
- [x] Examples appropriately added/updated.
- [x] Unit tests added/updated. See [DSC Community Testing Guidelines](https://dsccommunity.org/guidelines/testing-guidelines).
- [ ] Integration tests added/updated (where possible). See [DSC Community Testing Guidelines](https://dsccommunity.org/guidelines/testing-guidelines).
- [x] New/changed code adheres to [DSC Community Style Guidelines](https://dsccommunity.org/styleguidelines).

---
This change is [Reviewable](https://reviewable.io/reviews/dsccommunity/configmgrcbdsc/55)

### CODE REVIEW

File: source/DSCResources/DSC_CMStatusReportingComponent/DSC_CMStatusReportingComponent.psm1 [hunk 1]
```diff
@@ -0,0 +1,425 @@
...
+    )
+
+    Write-Verbose -Message $script:localizedData.RetrieveSettingValue
+    Import-ConfigMgrPowerShellModule -SiteCode $SiteCode
+    Set-Location -Path "$($SiteCode):\"
+
+    $clientComponentStatus = Get-CMStatusReportingComponent -SiteCode $SiteCode | Where-Object -FilterScript {$_.ItemName -eq 'Client Component Status Reporting'}
+    $serverComponentStatus = Get-CMStatusReportingComponent -SiteCode $SiteCode | Where-Object -FilterScript {$_.ItemName -eq 'Server Component Status Reporting'}
+
+    foreach ($prop in $clientComponentStatus.props)
```
Comment (jeffotterpohl) [inline source/DSCResources/DSC_CMStatusReportingComponent/DSC_CMStatusReportingComponent.psm1:35]: .Props

File: source/DSCResources/DSC_CMStatusReportingComponent/DSC_CMStatusReportingComponent.psm1 [hunk 2]
```diff
@@ -0,0 +1,425 @@
...
+                                                               'All'  { $cLogType = 'AllMilestonesAndAllDetails' }
+                                                               'EW'   { $cLogType = 'ErrorAndWarningMilestones' }
+                                                               'E'    { $cLogType = 'ErrorMilestones' }
+                                                           }
+                                                           $cLogFailure = [System.Convert]::ToBoolean($clientLogging[1])
+                                                       }
+        }
+    }
+
+    foreach ($prop in $serverComponentStatus.props)
```
Comment (jeffotterpohl) [inline source/DSCResources/DSC_CMStatusReportingComponent/DSC_CMStatusReportingComponent.psm1:70]: .Props

File: source/DSCResources/DSC_CMStatusReportingComponent/DSC_CMStatusReportingComponent.psm1 [hunk 3]
```diff
@@ -0,0 +1,425 @@
...
+        $ServerReportType
+    )
+
+    Import-ConfigMgrPowerShellModule -SiteCode $SiteCode
+    Set-Location -Path "$($SiteCode):\"
+    $state = Get-TargetResource -SiteCode $SiteCode
+
+    try
+    {
+        if ((($state.ClientLogChecked -eq $false) -and ($null -eq $ClientLogChecked) -and ($ClientLogType -or $ClientLogFailureChecked)) -or (($ClientLogChecked -eq $false) -and ($clientLogType -or $ClientLogFailureChecked)))
```
Comment (jeffotterpohl) [inline source/DSCResources/DSC_CMStatusReportingComponent/DSC_CMStatusReportingComponent.psm1:241]: check this value,  this is a boolean and would default to false

File: source/DSCResources/DSC_CMStatusReportingComponent/DSC_CMStatusReportingComponent.psm1 [hunk 4]
```diff
@@ -0,0 +1,425 @@
...
+        {
+            throw $script:localizedData.SReportThrow
+        }
+
+        $evalList = @('ClientLogChecked','ClientLogFailureChecked','ClientLogType','ClientReportChecked','ClientReportFailureChecked','ClientReportType'
+            'ServerLogChecked','ServerLogFailureChecked','ServerLogType','ServerReportChecked','ServerReportFailureChecked','ServerReportType')
+
+        foreach ($param in $PSBoundParameters.GetEnumerator())
+        {
+        if ($evalList -contains $param.key)
```
Comment (jeffotterpohl) [inline source/DSCResources/DSC_CMStatusReportingComponent/DSC_CMStatusReportingComponent.psm1:266]: Key

File: source/DSCResources/DSC_CMStatusReportingComponent/DSC_CMStatusReportingComponent.psm1 [hunk 5]
```diff
@@ -0,0 +1,425 @@
...
+            {
+                if ($param.Value -ne $state[$param.key])
```
Comment (jeffotterpohl) [inline source/DSCResources/DSC_CMStatusReportingComponent/DSC_CMStatusReportingComponent.psm1:268]: Key

File: source/DSCResources/DSC_CMStatusReportingComponent/DSC_CMStatusReportingComponent.psm1 (hunk 1 above)
Comment (NEllis280) [reply source/DSCResources/DSC_CMStatusReportingComponent/DSC_CMStatusReportingComponent.psm1]: Fixed

File: source/DSCResources/DSC_CMStatusReportingComponent/DSC_CMStatusReportingComponent.psm1 (hunk 2 above)
Comment (NEllis280) [reply source/DSCResources/DSC_CMStatusReportingComponent/DSC_CMStatusReportingComponent.psm1]: Fixed

File: source/DSCResources/DSC_CMStatusReportingComponent/DSC_CMStatusReportingComponent.psm1 (hunk 3 above)
Comment (NEllis280) [reply source/DSCResources/DSC_CMStatusReportingComponent/DSC_CMStatusReportingComponent.psm1]: Simplified throw statements

File: source/DSCResources/DSC_CMStatusReportingComponent/DSC_CMStatusReportingComponent.psm1 (hunk 4 above)
Comment (NEllis280) [reply source/DSCResources/DSC_CMStatusReportingComponent/DSC_CMStatusReportingComponent.psm1]: Fixed

File: source/DSCResources/DSC_CMStatusReportingComponent/DSC_CMStatusReportingComponent.psm1 (hunk 5 above)
Comment (NEllis280) [reply source/DSCResources/DSC_CMStatusReportingComponent/DSC_CMStatusReportingComponent.psm1]: Fixed

**Pull request merged**